Level.py

```diff
@@ -55,25 +56,35 @@ def define_level_properties(self):
     def get_touch_auth(self):
         for indx in range(self.ymax):
             for indy in range(self.xmax):
-                if (self.matrix[indx][indy][0] <= self.point[0] < self.matrix[indx][indy][2]) and (self.matrix[indx][indy][1] >= self.point[1] > self.matrix[indx][indy][3]):
-                    print(indx,indy,"ok")
-                    return
+                if (self.matrix[indy][indx][0] <= self.point[0] < self.matrix[indy][indx][2]) and (
+                                self.matrix[indy][indx][1] >= self.point[1] > self.matrix[indy][indx][3]):
+                    print(indy, indx, "ok")
+                    print(self.map_canvas.map_matrix[indy][indx])
+                    print(self.map_canvas.get_texture("A"))
+                    if self.map_canvas.map_matrix[indy][indx] == self.map_canvas.get_texture("A"):
+                        return True
+                    return False
 
     def on_touch_down(self, touch):
 
         # sauvegarde le touch
         ud = touch.ud
         ud['identifier'] = str(touch.uid)
 
-        # dessine le premier point
-        with self.canvas:
-            Color(0,0,1)
-            ud['points'] = [Point(points=(touch.x, touch.y), pointsize=self.touch_width, group = ud['identifier'])]
+        self.point = [touch.x, touch.y]
+
+        # dessine le premier point si l'emplacement est valide
+        if self.get_touch_auth():
```

Comment:
You should maybe rename the function called that the sentence makes sense when tested.
```python
if self.can_draw():
  pass
```
or 
```python
allowed_to_draw = self.can_draw()
if allowed_to_draw:
 pass
```